olp-cpp-sdk-core/src/network2/ios/OLPNetworkIOS.mm

```diff
@@ -0,0 +1,366 @@
+            unichar uc = [key characterAtIndex:i];
+            if ((uc >> 8))  // uc > 0xff
+              skey += (char)(uc >> 8);
+            skey += (char)uc;
+          }
+
+          NSString* value = response_headers[key];
+          std::string svalue;
+          for (int i = 0; i < value.length; ++i) {
+            unichar uc = [value characterAtIndex:i];
+            if ((uc >> 8))  // uc > 0xff
+              svalue += (char)(uc >> 8);
+            svalue += (char)uc;
+          }
+          header_callback(skey, svalue);
+        }
+      }
+    };
+
+    // setup handler for NSURLSessionDataTask::didReceiveData callback
+    task.dataHandler = ^(NSData* data) {
+      if (!weak_task) {
+        EDGE_SDK_LOG_WARNING_F(
+            kLogTag, "Data received after task=[%llu] was deleted!", requestId);
+        return;
+      }
+      __strong OLPHttpTask* strong_task = weak_task;
+      OLPHttpTaskResponseData* response_data = strong_task.responseData;
+      if (response_data.rangeOut) {
+        EDGE_SDK_LOG_TRACE(kLogTag, "Datacallback out of range");
+        return;
+      }
+
+      const size_t len = data.length;
+      if (data_callback) {
+        data_callback(reinterpret_cast<uint8_t*>(const_cast<void*>(data.bytes)),
+                      response_data.offset + response_data.count, len);
+      }
+
+      if (auto payload = strong_task.payload) {
+        if (payload->tellp() != std::streampos(response_data.count)) {
+          payload->seekp(response_data.count);
+          if (payload->fail()) {
+            EDGE_SDK_LOG_WARNING(
+                kLogTag,
+                "Reception stream doesn't support setting write point");
+            payload->clear();
+          }
+        }
+
+        const char* data_ptr = reinterpret_cast<const char*>(data.bytes);
+        payload->write(data_ptr, len);
+      }
+      response_data.count += len;
+    };
+
+    // setup handler for NSURLSessionDataTask::didCompleteWithError callback
+    task.completionHandler = ^(NSError* error) {
+      if (!weak_task) {
+        EDGE_SDK_LOG_WARNING_F(
+            kLogTag, "Completion received after task=[%llu] was deleted!",
+            requestId);
+        return;
+      }
+
+      __strong OLPHttpTask* strong_task = weak_task;
+      olp::network2::Network::Callback callback;
+      {
+        std::lock_guard<std::mutex> callback_lock(
+            *strong_task.callbackMutex.get());
+        if (strong_task.callback) {
+          callback = strong_task.callback;
+          strong_task.callback = nullptr;
+        }
+      }
+
+      OLPHttpTaskResponseData* response_data = strong_task.responseData;
+
+      // TODO: convert error code to string
+      std::string error_str = "Success";
+      int status = static_cast<int>(ErrorCode::NO_ERROR);
+      const bool cancelled = NSURLErrorCancelled == status;
+      if (callback) {
+        if (error && !cancelled) {
+          error_str = status < 0
```

Comment:
Can this be? `status` is set by you above to `NO_ERROR` and not modified. You only modify `status` on line 284.
Is this a mixup?

Reply:
Thanks for pointing that!